### PR TITLE
feat(charts): Adding chaos charts for node-memory-hog experiment

### PR DESCRIPTION
**Pull request by uditgaurav**
Signed-off-by: Udit Gaurav <uditgaurav@gmail.com>

**_This PR is for:_**

- Adding chaos charts for node-memory-hog experiment
- Removing extra permissions from node CPU hog experiment
**_Files Added:_**

```
charts/generic/node-memory-hog/engine.yaml
charts/generic/node-memory-hog/experiment.yaml
charts/generic/node-memory-hog/node-memory-hog.chartserviceversion.yaml
charts/generic/node-memory-hog/rbac.yaml
```

### CODE REVIEW

**Comment by uditgaurav** [comment]
**_Logs with rbac having minimum permission:_**

```
ansible-playbook 2.7.3
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python2.7/dist-packages/ansible
  executable location = /usr/local/bin/ansible-playbook
  python version = 2.7.17 (default, Nov  7 2019, 10:07:09) [GCC 7.4.0]
Using /etc/ansible/ansible.cfg as config file
/etc/ansible/hosts did not meet host_list requirements, check plugin documentation if this is unexpected
/etc/ansible/hosts did not meet script requirements, check plugin documentation if this is unexpected
statically imported: /experiments/generic/node_memory_hog/node_memory_hog_ansible_prerequisites.yml

2020-03-11 14:39:27.815805 PLAYBOOK: node_memory_hog_ansible_logic.yml 
1 plays in ./experiments/generic/node_memory_hog/node_memory_hog_ansible_logic.yml

2020-03-11 14:39:27.837361 ************ BRACE YOURSELF, EXPERIMENT BEGINS! ************ 

2020-03-11 14:39:33.451881 Step: Gathering Facts 
task path: /experiments/generic/node_memory_hog/node_memory_hog_ansible_logic.yml:2
META: ran handlers

2020-03-11 14:39:33.986221 Step: Identify the chaos util to be invoked 
task path: /experiments/generic/node_memory_hog/node_memory_hog_ansible_prerequisites.yml:1
details => {"changed": true, "checksum": "05311cb7e309713865c76d348dc9ad466dc28e52", "dest": "./chaosutil.yml", "gid": 0, "group": "root", "md5sum": "b2c1b1fb739f6064f372fbe57fbc3af2", "mode": "0644", "owner": "root", "size": 66, "src": "/root/.ansible/tmp/ansible-tmp-1583937573.51-157813942413522/source", "state": "file", "uid": 0}

2020-03-11 14:39:34.040815 Step: include_vars 
task path: /experiments/generic/node_memory_hog/node_memory_hog_ansible_logic.yml:25
details => {"ansible_facts": {"c_util": "/chaoslib/litmus/platform/gke/node_memory_consumption.yml"}, "ansible_included_var_files": ["/experiments/generic/node_memory_hog/chaosutil.yml"], "changed": false}
included: /utils/runtime/update_chaos_result_resource.yml for 127.0.0.1

2020-03-11 14:39:34.456572 Step: Generate the chaos result CR to reflect SOT (Start of Test) 
task path: /utils/runtime/update_chaos_result_resource.yml:3
details => {"changed": true, "checksum": "fb94345cb6cd6667e948944fd59bf2a05e5ee548", "dest": "/utils/runtime/chaos-result.yml", "gid": 0, "group": "root", "md5sum": "88c6dc1e961429854d46f3958fd89d26", "mode": "0644", "owner": "root", "size": 265, "src": "/root/.ansible/tmp/ansible-tmp-1583937574.2-162113440506781/source", "state": "file", "uid": 0}

2020-03-11 14:39:36.090024 Step: Apply the chaos result CR 
task path: /utils/runtime/update_chaos_result_resource.yml:13
details => {"changed": true, "cmd": "kubectl apply -f /utils/runtime/chaos-result.yml -n litmus", "delta": "0:00:01.197954", "end": "2020-03-11 14:39:36.063588", "failed_when_result": false, "rc": 0, "start": "2020-03-11 14:39:34.865634", "stderr": "", "stderr_lines": [], "stdout": "chaosresult.litmuschaos.io/node-memory-hog configured", "stdout_lines": ["chaosresult.litmuschaos.io/node-memory-hog configured"]}
included: /utils/common/status_app_pod.yml for 127.0.0.1

2020-03-11 14:39:37.144506 Step: Checking whether application pods are in running state 
task path: /utils/common/status_app_pod.yml:2
details => {"attempts": 1, "changed": true, "cmd": "kubectl get pods -n litmus -l run=nginx -o custom-columns=:.status.phase --no-headers", "delta": "0:00:00.733543", "end": "2020-03-11 14:39:37.115315", "rc": 0, "start": "2020-03-11 14:39:36.381772", "stderr": "", "stderr_lines": [], "stdout": "Running", "stdout_lines": ["Running"]}

2020-03-11 14:39:38.023828 Step: Get the container status of application. 
task path: /utils/common/status_app_pod.yml:9
details => {"attempts": 1, "changed": true, "cmd": "kubectl get pod -n litmus -l run=nginx --no-headers -o jsonpath='{.items[*].status.containerStatuses[*].ready}' | tr ' ' '\\n' | uniq", "delta": "0:00:00.681711", "end": "2020-03-11 14:39:37.992828", "rc": 0, "start": "2020-03-11 14:39:37.311117", "stderr": "", "stderr_lines": [], "stdout": "true", "stdout_lines": ["true"]}
included: /chaoslib/litmus/platform/gke/node_memory_consumption.yml for 127.0.0.1

2020-03-11 14:39:39.163923 Step: Getting the pre-chaos application pod name 
task path: /chaoslib/litmus/platform/gke/node_memory_consumption.yml:4
details => {"changed": false, "resources": [{"apiVersion": "v1", "kind": "Pod", "metadata": {"creationTimestamp": "2020-03-11T14:30:25Z", "generateName": "nginx-7db9fccd9b-", "labels": {"pod-template-hash": "7db9fccd9b", "run": "nginx"}, "name": "nginx-7db9fccd9b-tr9b4", "namespace": "litmus", "ownerReferences": [{"apiVersion": "apps/v1", "blockOwnerDeletion": true, "controller": true, "kind": "ReplicaSet", "name": "nginx-7db9fccd9b", "uid": "d921ba16-63a4-11ea-bb89-42010a80003a"}], "resourceVersion": "356829", "selfLink": "/api/v1/namespaces/litmus/pods/nginx-7db9fccd9b-tr9b4", "uid": "d925ed59-63a4-11ea-bb89-42010a80003a"}, "spec": {"containers": [{"image": "nginx", "imagePullPolicy": "Always", "name": "nginx", "resources": {}, "terminationMessagePath": "/dev/termination-log", "terminationMessagePolicy": "File", "volumeMounts": [{"mountPath": "/var/run/secrets/kubernetes.io/serviceaccount", "name": "default-token-8w2pv", "readOnly": true}]}], "dnsPolicy": "ClusterFirst", "enableServiceLinks": true, "nodeName": "gke-udit-limtus-e2e-default-pool-ec4ec958-3k9b", "priority": 0, "restartPolicy": "Always", "schedulerName": "default-scheduler", "securityContext": {}, "serviceAccount": "default", "serviceAccountName": "default", "terminationGracePeriodSeconds": 30, "tolerations": [{"effect": "NoExecute", "key": "node.kubernetes.io/not-ready", "operator": "Exists", "tolerationSeconds": 300}, {"effect": "NoExecute", "key": "node.kubernetes.io/unreachable", "operator": "Exists", "tolerationSeconds": 300}], "volumes": [{"name": "default-token-8w2pv", "secret": {"defaultMode": 420, "secretName": "default-token-8w2pv"}}]}, "status": {"conditions": [{"lastProbeTime": null, "lastTransitionTime": "2020-03-11T14:30:25Z", "status": "True", "type": "Initialized"}, {"lastProbeTime": null, "lastTransitionTime": "2020-03-11T14:30:27Z", "status": "True", "type": "Ready"}, {"lastProbeTime": null, "lastTransitionTime": "2020-03-11T14:30:27Z", "status": "True", "type": "ContainersReady"}, {"lastProbeTime": null, "lastTransitionTime": "2020-03-11T14:30:25Z", "status": "True", "type": "PodScheduled"}], "containerStatuses": [{"containerID": "docker://8c4110b90eade9d0b7e47b3ef882c96a712d0a758b93ac4f3eaa79813bdd1b55", "image": "nginx:latest", "imageID": "docker-pullable://nginx@sha256:3936fb3946790d711a68c58be93628e43cbca72439079e16d154b5db216b58da", "lastState": {}, "name": "nginx", "ready": true, "restartCount": 0, "state": {"running": {"startedAt": "2020-03-11T14:30:27Z"}}}], "hostIP": "10.128.0.39", "phase": "Running", "podIP": "10.20.1.51", "qosClass": "BestEffort", "startTime": "2020-03-11T14:30:25Z"}}]}

2020-03-11 14:39:39.236147 Step: Recording the node name of application pod 
task path: /chaoslib/litmus/platform/gke/node_memory_consumption.yml:12
details => {"ansible_facts": {"node_name": "gke-udit-limtus-e2e-default-pool-ec4ec958-3k9b"}, "changed": false}

2020-03-11 14:39:40.067503 Step: Generate a run id if not passed from the engine/experiment 
task path: /chaoslib/litmus/platform/gke/node_memory_consumption.yml:17
details => {"changed": true, "cmd": "echo $(mktemp) | cut -d '.' -f 2", "delta": "0:00:00.612802", "end": "2020-03-11 14:39:40.043887", "rc": 0, "start": "2020-03-11 14:39:39.431085", "stderr": "", "stderr_lines": [], "stdout": "lTUvtkH5xU", "stdout_lines": ["lTUvtkH5xU"]}

2020-03-11 14:39:40.131920 Step: set_fact 
task path: /chaoslib/litmus/platform/gke/node_memory_consumption.yml:21
details => {"ansible_facts": {"run_id": "ltuvtkh5xu"}, "changed": false}

2020-03-11 14:39:40.521418 Step: Patch the engine uid 
task path: /chaoslib/litmus/platform/gke/node_memory_consumption.yml:35
details => {"changed": true, "checksum": "1c72cd18b2bc19cf45a6f180c09e903b0a0421cc", "dest": "/chaoslib/litmus/platform/gke/node-memory-hog-job.yml", "gid": 0, "group": "root", "md5sum": "f6565aec7583282282de03ca6602e60e", "mode": "0644", "owner": "root", "size": 505, "src": "/root/.ansible/tmp/ansible-tmp-1583937580.23-122987132024276/source", "state": "file", "uid": 0}

2020-03-11 14:39:41.437127 Step: Creating Job for memory-hog 
task path: /chaoslib/litmus/platform/gke/node_memory_consumption.yml:49
details => {"changed": true, "method": "create", "result": {"apiVersion": "batch/v1", "kind": "Job", "metadata": {"creationTimestamp": "2020-03-11T14:39:41Z", "labels": {"chaosUID": ""}, "name": "node-memory-hog-ltuvtkh5xu", "namespace": "litmus", "resourceVersion": "359836", "selfLink": "/apis/batch/v1/namespaces/litmus/jobs/node-memory-hog-ltuvtkh5xu", "uid": "2457dcc6-63a6-11ea-bb89-42010a80003a"}, "spec": {"backoffLimit": 6, "completions": 1, "parallelism": 1, "selector": {"matchLabels": {"controller-uid": "2457dcc6-63a6-11ea-bb89-42010a80003a"}}, "template": {"metadata": {"creationTimestamp": null, "labels": {"app": "memory-hog", "chaosUID": "", "controller-uid": "2457dcc6-63a6-11ea-bb89-42010a80003a", "job-name": "node-memory-hog-ltuvtkh5xu"}}, "spec": {"containers": [{"args": ["--vm", "1", "--vm-bytes", "90%", "-t", "60s"], "image": "alexeiled/stress-ng:latest", "imagePullPolicy": "Always", "name": "memory-hog", "resources": {}, "terminationMessagePath": "/dev/termination-log", "terminationMessagePolicy": "File"}], "dnsPolicy": "ClusterFirst", "nodeSelector": {"kubernetes.io/hostname": "gke-udit-limtus-e2e-default-pool-ec4ec958-3k9b"}, "restartPolicy": "Never", "schedulerName": "default-scheduler", "securityContext": {}, "terminationGracePeriodSeconds": 30}}}, "status": {}}}
FAILED - RETRYING: Checking the status of Job (120 retries left).
 [WARNING]: when statements should not include jinja2 templating delimiters
such as {{ }} or {% %}. Found: {{ job_pod |
json_query('resources[0].status.phase') == 'Succeeded' }}
FAILED - RETRYING: Checking the status of Job (119 retries left).
FAILED - RETRYING: Checking the status of Job (118 retries left).
FAILED - RETRYING: Checking the status of Job (117 retries left).
FAILED - RETRYING: Checking the status of Job (116 retries left).
FAILED - RETRYING: Checking the status of Job (115 retries left).
FAILED - RETRYING: Checking the status of Job (114 retries left).
FAILED - RETRYING: Checking the status of Job (113 retries left).
FAILED - RETRYING: Checking the status of Job (112 retries left).
FAILED - RETRYING: Checking the status of Job (111 retries left).
FAILED - RETRYING: Checking the status of Job (110 retries left).
FAILED - RETRYING: Checking the status of Job (109 retries left).
FAILED - RETRYING: Checking the status of Job (108 retries left).
FAILED - RETRYING: Checking the status of Job (107 retries left).
FAILED - RETRYING: Checking the status of Job (106 retries left).
FAILED - RETRYING: Checking the status of Job (105 retries left).
FAILED - RETRYING: Checking the status of Job (104 retries left).
FAILED - RETRYING: Checking the status of Job (103 retries left).
FAILED - RETRYING: Checking the status of Job (102 retries left).
FAILED - RETRYING: Checking the status of Job (101 retries left).
FAILED - RETRYING: Checking the status of Job (100 retries left).
FAILED - RETRYING: Checking the status of Job (99 retries left).
FAILED - RETRYING: Checking the status of Job (98 retries left).
FAILED - RETRYING: Checking the status of Job (97 retries left).
FAILED - RETRYING: Checking the status of Job (96 retries left).
FAILED - RETRYING: Checking the status of Job (95 retries left).
FAILED - RETRYING: Checking the status of Job (94 retries left).
FAILED - RETRYING: Checking the status of Job (93 retries left).
FAILED - RETRYING: Checking the status of Job (92 retries left).
FAILED - RETRYING: Checking the status of Job (91 retries left).
FAILED - RETRYING: Checking the status of Job (90 retries left).
FAILED - RETRYING: Checking the status of Job (89 retries left).
FAILED - RETRYING: Checking the status of Job (88 retries left).
FAILED - RETRYING: Checking the status of Job (87 retries left).
FAILED - RETRYING: Checking the status of Job (86 retries left).
FAILED - RETRYING: Checking the status of Job (85 retries left).
FAILED - RETRYING: Checking the status of Job (84 retries left).

2020-03-11 14:40:44.968266 Step: Checking the status of Job 
task path: /chaoslib/litmus/platform/gke/node_memory_consumption.yml:57
details => {"attempts": 38, "changed": false, "resources": [{"apiVersion": "v1", "kind": "Pod", "metadata": {"creationTimestamp": "2020-03-11T14:39:41Z", "generateName": "node-memory-hog-ltuvtkh5xu-", "labels": {"app": "memory-hog", "chaosUID": "", "controller-uid": "2457dcc6-63a6-11ea-bb89-42010a80003a", "job-name": "node-memory-hog-ltuvtkh5xu"}, "name": "node-memory-hog-ltuvtkh5xu-qd8n7", "namespace": "litmus", "ownerReferences": [{"apiVersion": "batch/v1", "blockOwnerDeletion": true, "controller": true, "kind": "Job", "name": "node-memory-hog-ltuvtkh5xu", "uid": "2457dcc6-63a6-11ea-bb89-42010a80003a"}], "resourceVersion": "360171", "selfLink": "/api/v1/namespaces/litmus/pods/node-memory-hog-ltuvtkh5xu-qd8n7", "uid": "24595342-63a6-11ea-bb89-42010a80003a"}, "spec": {"containers": [{"args": ["--vm", "1", "--vm-bytes", "90%", "-t", "60s"], "image": "alexeiled/stress-ng:latest", "imagePullPolicy": "Always", "name": "memory-hog", "resources": {}, "terminationMessagePath": "/dev/termination-log", "terminationMessagePolicy": "File", "volumeMounts": [{"mountPath": "/var/run/secrets/kubernetes.io/serviceaccount", "name": "default-token-8w2pv", "readOnly": true}]}], "dnsPolicy": "ClusterFirst", "enableServiceLinks": true, "nodeName": "gke-udit-limtus-e2e-default-pool-ec4ec958-3k9b", "nodeSelector": {"kubernetes.io/hostname": "gke-udit-limtus-e2e-default-pool-ec4ec958-3k9b"}, "priority": 0, "restartPolicy": "Never", "schedulerName": "default-scheduler", "securityContext": {}, "serviceAccount": "default", "serviceAccountName": "default", "terminationGracePeriodSeconds": 30, "tolerations": [{"effect": "NoExecute", "key": "node.kubernetes.io/not-ready", "operator": "Exists", "tolerationSeconds": 300}, {"effect": "NoExecute", "key": "node.kubernetes.io/unreachable", "operator": "Exists", "tolerationSeconds": 300}], "volumes": [{"name": "default-token-8w2pv", "secret": {"defaultMode": 420, "secretName": "default-token-8w2pv"}}]}, "status": {"conditions": [{"lastProbeTime": null, "lastTransitionTime": "2020-03-11T14:39:41Z", "reason": "PodCompleted", "status": "True", "type": "Initialized"}, {"lastProbeTime": null, "lastTransitionTime": "2020-03-11T14:40:44Z", "reason": "PodCompleted", "status": "False", "type": "Ready"}, {"lastProbeTime": null, "lastTransitionTime": "2020-03-11T14:40:44Z", "reason": "PodCompleted", "status": "False", "type": "ContainersReady"}, {"lastProbeTime": null, "lastTransitionTime": "2020-03-11T14:39:41Z", "status": "True", "type": "PodScheduled"}], "containerStatuses": [{"containerID": "docker://713d7c31364909a1314339e20397cd70fc7703138863e91c01521597b7ef833b", "image": "alexeiled/stress-ng:latest", "imageID": "docker-pullable://alexeiled/stress-ng@sha256:51adb68b679b6ec424729482d9a0c6fe8eb74a5c030a854f96e5dc14122d99c7", "lastState": {}, "name": "memory-hog", "ready": false, "restartCount": 0, "state": {"terminated": {"containerID": "docker://713d7c31364909a1314339e20397cd70fc7703138863e91c01521597b7ef833b", "exitCode": 0, "finishedAt": "2020-03-11T14:40:43Z", "reason": "Completed", "startedAt": "2020-03-11T14:39:43Z"}}}], "hostIP": "10.128.0.39", "phase": "Succeeded", "podIP": "10.20.1.55", "qosClass": "BestEffort", "startTime": "2020-03-11T14:39:41Z"}}]}

2020-03-11 14:40:45.922363 Step: Getting the pod logs 
task path: /chaoslib/litmus/platform/gke/node_memory_consumption.yml:69
details => {"changed": true, "cmd": "kubectl logs node-memory-hog-ltuvtkh5xu-qd8n7 -n litmus", "delta": "0:00:00.714067", "end": "2020-03-11 14:40:45.896386", "rc": 0, "start": "2020-03-11 14:40:45.182319", "stderr": "", "stderr_lines": [], "stdout": "stress-ng: info:  [1] dispatching hogs: 1 vm\nstress-ng: info:  [1] successful run completed in 60.06s (1 min, 0.06 secs)", "stdout_lines": ["stress-ng: info:  [1] dispatching hogs: 1 vm", "stress-ng: info:  [1] successful run completed in 60.06s (1 min, 0.06 secs)"]}

2020-03-11 14:40:46.884007 Step: Getting the node status 
task path: /chaoslib/litmus/platform/gke/node_memory_consumption.yml:80
details => {"changed": true, "cmd": "kubectl get nodes gke-udit-limtus-e2e-default-pool-ec4ec958-3k9b | awk '{print $2}' | tail -n1", "delta": "0:00:00.683769", "end": "2020-03-11 14:40:46.823947", "rc": 0, "start": "2020-03-11 14:40:46.140178", "stderr": "", "stderr_lines": [], "stdout": "Ready", "stdout_lines": ["Ready"]}

2020-03-11 14:40:47.870234 Step: Deleting the Memory Hog Job 
task path: /chaoslib/litmus/platform/gke/node_memory_consumption.yml:85
details => {"changed": true, "cmd": "kubectl delete -f /chaoslib/litmus/platform/gke/node-memory-hog-job.yml -n litmus", "delta": "0:00:00.710170", "end": "2020-03-11 14:40:47.847629", "rc": 0, "start": "2020-03-11 14:40:47.137459", "stderr": "", "stderr_lines": [], "stdout": "job.batch \"node-memory-hog-ltuvtkh5xu\" deleted", "stdout_lines": ["job.batch \"node-memory-hog-ltuvtkh5xu\" deleted"]}

2020-03-11 14:40:48.576549 Step: Confirm that the Memory Hog Job is deleted successfully 
task path: /chaoslib/litmus/platform/gke/node_memory_consumption.yml:91
details => {"attempts": 1, "changed": false, "resources": []}
included: /utils/common/status_app_pod.yml for 127.0.0.1

2020-03-11 14:40:49.663922 Step: Checking whether application pods are in running state 
task path: /utils/common/status_app_pod.yml:2
details => {"attempts": 1, "changed": true, "cmd": "kubectl get pods -n litmus -l run=nginx -o custom-columns=:.status.phase --no-headers", "delta": "0:00:00.707990", "end": "2020-03-11 14:40:49.636686", "rc": 0, "start": "2020-03-11 14:40:48.928696", "stderr": "", "stderr_lines": [], "stdout": "Running", "stdout_lines": ["Running"]}

2020-03-11 14:40:50.553150 Step: Get the container status of application. 
task path: /utils/common/status_app_pod.yml:9
details => {"attempts": 1, "changed": true, "cmd": "kubectl get pod -n litmus -l run=nginx --no-headers -o jsonpath='{.items[*].status.containerStatuses[*].ready}' | tr ' ' '\\n' | uniq", "delta": "0:00:00.682125", "end": "2020-03-11 14:40:50.522509", "rc": 0, "start": "2020-03-11 14:40:49.840384", "stderr": "", "stderr_lines": [], "stdout": "true", "stdout_lines": ["true"]}

2020-03-11 14:40:50.829197 Step: set_fact 
task path: /experiments/generic/node_memory_hog/node_memory_hog_ansible_logic.yml:144
details => {"ansible_facts": {"flag": "Pass"}, "changed": false}
included: /utils/runtime/getting_failure_step.yml for 127.0.0.1

2020-03-11 14:40:51.209492 Step: Record the name of failed step 
task path: /utils/runtime/getting_failure_step.yml:33
details => {"ansible_facts": {"failStep": "N/A"}, "changed": false}
included: /utils/runtime/update_chaos_result_resource.yml for 127.0.0.1

2020-03-11 14:40:52.315693 Step: Update the chaos result CR to reflect EOT (End of Test) 
task path: /utils/runtime/update_chaos_result_resource.yml:23
details => {"changed": true, "checksum": "15c830e1702fc58af93a8513967a99a242560a58", "dest": "/utils/runtime/chaos-result.yml", "gid": 0, "group": "root", "md5sum": "588c4885274f3fc29a32a0fd772c53fd", "mode": "0644", "owner": "root", "size": 282, "src": "/root/.ansible/tmp/ansible-tmp-1583937651.48-91879257625977/source", "state": "file", "uid": 0}

2020-03-11 14:40:53.375181 Step: Apply the chaos result CR 
task path: /utils/runtime/update_chaos_result_resource.yml:33
details => {"changed": true, "cmd": "kubectl apply -f /utils/runtime/chaos-result.yml -n litmus", "delta": "0:00:00.834767", "end": "2020-03-11 14:40:53.339010", "failed_when_result": false, "rc": 0, "start": "2020-03-11 14:40:52.504243", "stderr": "", "stderr_lines": [], "stdout": "chaosresult.litmuschaos.io/node-memory-hog configured", "stdout_lines": ["chaosresult.litmuschaos.io/node-memory-hog configured"]}
META: ran handlers
META: ran handlers

2020-03-11 14:40:53.382262 ************ RELAX, EXPERIMENT ENDS! ************ 
127.0.0.1                  : ok=30   changed=15   unreachable=0    failed=0   
```

File: charts/generic/node-memory-hog/experiment.yaml [hunk 1]
```diff
@@ -0,0 +1,66 @@
+apiVersion: litmuschaos.io/v1alpha1
+description:
+  message: |
+    Give a memory hog on a node belonging to a deployment
+kind: ChaosExperiment
+metadata:
+  name: node-memory-hog
+  version: 0.0.1
```
Comment (ispeakc0de) [inline charts/generic/node-memory-hog/experiment.yaml]: 0.1.0

File: charts/generic/node-memory-hog/experiment.yaml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  definition:
+    scope: Cluster
+    permissions:
+      - apiGroups:
+          - ""
+          - "batch"
+          - "apps"
+          - "litmuschaos.io"
+        resources:
+          - "daemonsets"
```
Comment (ispeakc0de) [inline charts/generic/node-memory-hog/experiment.yaml]: remove this

File: charts/generic/node-memory-hog/experiment.yaml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+    # Period to wait before injection of chaos in sec
+    - name: RAMP_TIME
+      value: ''
+
+    - name: PLATFORM
+      value: 'GKE'
+
+    - name: LIB
+      value: ''
```
Comment (ispeakc0de) [inline charts/generic/node-memory-hog/experiment.yaml]: pls provide default value

File: charts/generic/node-memory-hog/node-memory-hog.chartserviceversion.yaml [hunk 4]
```diff
@@ -0,0 +1,38 @@
+apiVersion: litmuchaos.io/v1alpha1
+kind: ChartServiceVersion
+metadata:
+  createdAt: 2020-03-28T10:28:08Z
+  name: node-cpu-hog
```
Comment (ispeakc0de) [inline charts/generic/node-memory-hog/node-memory-hog.chartserviceversion.yaml]: node-memory-hog

File: charts/generic/node-memory-hog/node-memory-hog.chartserviceversion.yaml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  version: 0.0.1
```
Comment (ispeakc0de) [inline charts/generic/node-memory-hog/node-memory-hog.chartserviceversion.yaml]: 0.1.0

File: charts/generic/node-memory-hog/node-memory-hog.chartserviceversion.yaml [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+  annotations:
+    categories: Kubernetes
+    vendor: CNCF
+    support: https://slack.kubernetes.io/
+spec:
+  displayName: node-memory-hog
+  categoryDescription: |
+    Kubernetes Node memory hog contains chaos to disrupt state of kubernetes resources. Experiments can inject a memory skipe on a node where the application pod is scheduled.
```
Comment (ispeakc0de) [inline charts/generic/node-memory-hog/node-memory-hog.chartserviceversion.yaml]: skipe ?

File: charts/generic/node-memory-hog/node-memory-hog.chartserviceversion.yaml [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    - Causes (forced/graceful) Memory hog on a perticular node where the application deployment is available.
+    - After test the recovery should be manual of the application pod and node in case they are not in appropriate state. The chances of such scenarios is when the memory percentance provided is greater than 100%.
+ 
+  keywords:
+    - Kubernetes
+    - Memory
+    - State
```
Comment (ispeakc0de) [inline charts/generic/node-memory-hog/node-memory-hog.chartserviceversion.yaml:21]: node

File: charts/generic/node-memory-hog/node-memory-hog.chartserviceversion.yaml [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+  keywords:
+    - Kubernetes
+    - Memory
+    - State 
+  platforms:
+    - GKE
+  maturity: alpha
+  chaosType: infra
+  maintainers:
+    - name: udit
```
Comment (ispeakc0de) [inline charts/generic/node-memory-hog/node-memory-hog.chartserviceversion.yaml]: Udit Gaurav

File: charts/generic/node-memory-hog/rbac.yaml [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+---
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRole
+metadata:
+  name: node-memory-hog-sa
+  labels:
+    name: node-memory-hog-sa
+rules:
+- apiGroups: ["","litmuschaos.io","batch","apps"]
+  resources: ["pods","daemonsets","jobs","pods/log","events","chaosengines","chaosexperiments","chaosresults"]
```
Comment (ispeakc0de) [inline charts/generic/node-memory-hog/rbac.yaml]: daemonsets?

**Review by ispeakc0de — CHANGES_REQUESTED** [review]

File: charts/generic/node-memory-hog/node-memory-hog.chartserviceversion.yaml [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+    - Kubernetes
+    - Memory
+    - State 
+    - Node
+  platforms:
+    - GKE
+  maturity: alpha
+  chaosType: infra
+  maintainers:
+    - name: Udit Gaurav
```
Comment (ispeakc0de) [inline charts/generic/node-memory-hog/node-memory-hog.chartserviceversion.yaml:28]: :+1:

**Review by ispeakc0de — DISMISSED** [review]
/lgtm

File: charts/generic/node-memory-hog/node-memory-hog.chartserviceversion.yaml [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+  version: 0.1.0
+  annotations:
+    categories: Kubernetes
+    vendor: CNCF
+    support: https://slack.kubernetes.io/
+spec:
+  displayName: node-memory-hog
+  categoryDescription: |
+    Kubernetes Node memory hog contains chaos to disrupt state of kubernetes resources. Experiments can inject a memory spike on a node where the application pod is scheduled.
+    - Causes (forced/graceful) Memory hog on a perticular node where the application deployment is available.
```
Comment (ksatchit) [inline charts/generic/node-memory-hog/node-memory-hog.chartserviceversion.yaml]: there is no such thing as a force/graceful memory hog. Also do a spell check

File: charts/generic/node-memory-hog/node-memory-hog.chartserviceversion.yaml [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+    - After test the recovery should be manual of the application pod and node in case they are not in appropriate state. The chances of such scenarios is when the memory percentance provided is greater than 100%.
```
Comment (ksatchit) [inline charts/generic/node-memory-hog/node-memory-hog.chartserviceversion.yaml]: Look at the recovery statement used in CPU hog test

**Review by ksatchit — CHANGES_REQUESTED** [review]
description changes needed in csv

**Comment by ispeakc0de** [comment]
@uditgaurav  Can you please add this experiment to package, generic CSV, and an image for this.

**Review by ispeakc0de — APPROVED** [review]
/lgtm

**Review by ksatchit — APPROVED** [review]

**Pull request merged**